app/models/ArrivalUpdate.scala

```diff
@@ -59,14 +60,15 @@ object ArrivalUpdate {
 final case class MessageStatusUpdate(messageId: MessageId, messageStatus: MessageStatus) extends ArrivalUpdate
 
 object MessageStatusUpdate extends MongoDateTimeFormats {
-  implicit def arrivalStateUpdate(implicit writes: Writes[MessageStatus]): ArrivalModifier[MessageStatusUpdate] =
+
+  implicit def arrivalStateUpdate(implicit clock: Clock, writes: Writes[MessageStatus]): ArrivalModifier[MessageStatusUpdate] =
     ArrivalModifier(
       value =>
         Json.obj(
           "$set" ->
             Json.obj(
               s"messages.${value.messageId.index}.status" -> value.messageStatus,
-              "lastUpdated"                               -> LocalDateTime.now.withSecond(0).withNano(0)
+              "lastUpdated"                               -> LocalDateTime.now(clock).withSecond(0).withNano(0)
```

Comment:
This use of `LocalDateTime.now` was causing nondeterministic failures in the test suite.

I'm not wild about plumbing a `Clock` through everything implicitly but there's little other choice

Reply:
I think the withSecond and withNano can be removed now and an appropriate clock can be used in the tests